File: Strings-letters comparison.c
#include<stdio.h>
#include <stdlib.h>

int main()
{
char s1[100],s2[100];
scanf("%s\n%s",s1,s2);
for(int i=0;i<strlen(s1);i++)
{
    for(int j=i+1;j<strlen(s1);j++)
    {
        if(s1[i]==s1[j])
        {
            s1[j]='\0';
        }
    }
}
for(int i=0;i<strlen(s2);i++)
{
    for(int j=i+1;j<strlen(s2);j++)
    {
        if(s2[i]==s2[j])
        {
            s2[j]='\0';
        }
    }
}
int count=0;
for(int i=0;i<strlen(s1);i++)
{
    for(int j=0;j<strlen(s2);j++)
    {
        if(s1[i]==s2[j])
        {
            count++;
        }
    }
}


if(count==strlen(s1)||count==strlen(s2))
{
    printf("YES");
}
else
{
    printf("NO");

}
}

File: Strings-letters comparison.py
"""
INPUT:
read
dear
OUTPUT:
YES
INPUT:
record
decoder
OUTPUT:
YES
INPUT:
energy
synergy
OUTPUT:
NO
"""
s=input().strip()
s1=input().strip()
s=list(s)
s1=list(s1)
for i in range(0,len(s)):
    for j in range(i+1,len(s)):
        if s[i]==s[j]:
            s[j]=""
for i in range(0,len(s1)):
    for j in range(i+1,len(s1)):
        if s1[i]==s1[j]:
           s1[j]=""
count=0
for i in range(0,len(s)):
    if s[i] in s1:
        count+=1
if count==len(s):
    print("YES")
else:
    print("NO")
